carla/models/catalog/catalog.py

```diff
@@ -190,9 +190,30 @@ def predict(self, x):
         input = self.perform_pipeline(x) if self._use_pipeline else x
 
         if self._backend == "pytorch":
-            return self._model.predict(input)
+            # Pytorch model needs torch.Tensor as input
+            if torch.is_tensor(input):
+                device = "cuda" if input.is_cuda else "cpu"
```

Comment:
For my understanding what is `input` here and how did it get the `is_cuda` property? i thought `input` comes from `x` and is the data we want to predict?

Reply:
Since we have to handle torch input now too, a user could decide to work on cpu or gpu. Depending on that, we have to move our pre-trained model and input to the same level.

Reply:
Makes sense. So now x/inputs can also be pytorch sensors and with these we need to move them to the same device as the model. Cool 👍🏽